main.py

```diff
@@ -120,36 +122,46 @@ def delete_Pokemon(pokemon_id):
                             detail="Pokemon with this id is either alreday deleted or not found..")
 
 
-@app.post("/fetch-and-store/")
+@app.post("/pokemon/fetch_and_store/")
 def fetch_and_store():
     response = requests.get('https://coralvanda.github.io/pokemon_data.json')
     data = response.json()
     print(f"{data}- data fetched")
 
-    for i in range(len(data)):
-        for j in range(i + 1, len(data)):
-            while data[i]["#"] == data[j]["#"]:
-                data[j]["#"] += 1
+    # Adjust Pokémon IDs to ensure unique
+    ids = set()
+    for pokemon in data:
+        current_id = pokemon["#"]
+        while current_id in ids:
+            current_id += 1
```

Comment:
instead of while loop you can do `current_id = max(ids) + 1`

Reply:
you can re-factor this entire id logic as below (assuming that we get list of pokemon data in sorted IDs)

```python
max_id = 0
for pokemon in data:
    pokemon["#"] = max_id+1
    max_id+=1
``` 

if we don't get pokemon in sorted IDs, then 
```python
ids = set()
for pokemon in data:
    if pokemon["#"] in ids:
        pokemon["#"] = max(ids) + 1
        ids.add(pokemon["#"])
``` 

don't create variables even for small tasks unnecessarily.
using fewer variables will improve memory management.